app/process.py

```diff
@@ -86,6 +87,52 @@ def diff_texts(text1, text2):
     return highlighted_text
 
 
+def tokenize_mixed_direction_text(text: str, language: str) -> str:
+    """
+    Tokenizes a given text while correctly handling the embedding of LTR text within an RTL context,
+    preserving formatting such as spaces and new lines. LTR words are isolated using Unicode
+    Bidirectional Algorithm control characters to ensure proper display within RTL text.
+
+    Args:
+        text (str): The text to be tokenized and formatted.
+        language (str): The language of the text, which determines text directionality.
+
+    Returns:
+        str: The tokenized text with LTR words appropriately wrapped to preserve reading flow in RTL languages.
+    """
+    rtl_languages = {
+        "Arabic",
+        "Aramaic",
+        "Azeri",
+        "Dhivehi/Maldivian",
+        "Hebrew",
+        "Kurdish (Sorani)",
+        "Persian/Farsi",
+        "Urdu",
+    }
+    is_rtl = language in rtl_languages
+
+    # Regex to capture words, non-word characters, and any whitespace
+    words_and_delimiters = re.findall(r"\w+|[^\w\s]+|\s+", text)
+
+    new_text = []
+    ltr_pattern = re.compile(

```

Comment:
@Salah-Sal 
Thanks for your contribution.
can we modify the ltr_pattern to include numerical digits?
 
```python
ltr_pattern = re.compile("[A-Za-z0-9]+")
```